src/Polly.Core/ResilienceStrategyExtensions.Async.Task.cs

```diff
@@ -0,0 +1,72 @@
+using System.Threading;
+using Polly;
+
+namespace Polly;
+
+public static partial class ResilienceStrategyExtensions
+{
+    /// <summary>
+    /// Executes the <paramref name="execution"/> callback.
+    /// </summary>
+    /// <typeparam name="TState">The type of state associated with the execution.</typeparam>
+    /// <param name="strategy">Instance of <see cref="IResilienceStrategy"/>.</param>
+    /// <param name="execution">The execution callback.</param>
+    /// <param name="context">The context associated with the execution.</param>
+    /// <param name="state">The state associated with the execution.</param>
+    /// <returns>The instance of <see cref="Task"/> that represents the asynchronous execution.</returns>
```

Comment:
Should we also document the exceptions that would be thrown by methods using the `Guard` methods?

Reply:
I am split on this. I feel it just adds noise to the documentation. If you think we need this I can add it everywhere.

Reply:
I'm happy to not, I just thought I'd suggest it.